crowbar_framework/lib/crowbar/chef/upload.rb

```diff
@@ -227,15 +227,12 @@ def local_cookbook_md5_checksums(cookbook)
       end
 
       def cookbook_exist?(cookbook)
-        local_md5sums = local_cookbook_md5_checksums(cookbook)
-        remote_md5sums = api_cookbook_md5_checksums(cookbook.name)
-        combined_md5sums = local_md5sums.deep_merge(remote_md5sums)
-
-        combined_md5sums.reject! { |cb, _| local_md5sums.include? cb }
-        return false unless combined_md5sums.empty?
-
-        logger.info("Cookbook #{cookbook.name} already exists, skipping...")
-        true
+        if local_cookbook_md5_checksums(cookbook) == api_cookbook_md5_checksums(cookbook.name)
```

Comment:
Nop, but when I wrote the code I tough way too complex. I did forgot that it's possible to compare hashs in ruby without caring about the order. I tried to implement something which checks if a hash is the same but `deep_merge` did not work as expected so at the end it was always true.